streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -725,27 +727,79 @@ public boolean isProcessable(final long wallClockTime) {
             // a task is only closing / closed when 1) task manager is closing, 2) a rebalance is undergoing;
             // in either case we can just log it and move on without notifying the thread since the consumer
             // would soon be updated to not return any records for this task anymore.
-            log.info("Stream task {} is already in {} state, skip processing it.", id(), state());
+            log.info("Task is already in {} state, skip processing it.", state());
 
             return false;
         }
 
+        final boolean wasReady = lastNotReadyLogTime == -1L;
         if (hasPendingTxCommit) {
             // if the task has a pending TX commit, we should just retry the commit but not process any records
             // thus, the task is not processable, even if there is available data in the record queue
+            if (wasReady) {
+                // READY -> NOT_READY - start timer
+                lastNotReadyLogTime = wallClockTime;
+            } else {
+                // NOT_READY - check if it should log
+                final long timeSinceLastLog = wallClockTime - lastNotReadyLogTime;
+                if (timeSinceLastLog >= NOT_READY_LOG_INTERVAL_MS) {
+                    log.info("Task is not ready to process: has pending transaction commit");
+                    lastNotReadyLogTime = wallClockTime;
+                }
+            }
             return false;
         }
+
         final boolean readyToProcess = partitionGroup.readyToProcess(wallClockTime);
         if (!readyToProcess) {
-            if (timeCurrentIdlingStarted.isEmpty()) {
+            if (wasReady) {
+                // READY -> NOT_READY - start the timer
+                lastNotReadyLogTime = wallClockTime;
                 timeCurrentIdlingStarted = Optional.of(wallClockTime);
+            } else {
+                // NOT_READY - check if it should log
+                final long timeSinceLastLog = wallClockTime - lastNotReadyLogTime;
+                if (timeSinceLastLog >= NOT_READY_LOG_INTERVAL_MS) {
+                    final String partitionStatus = getNotReadyPartitionStatus();
```

Comment:
If we really get the details from `partitionGroup.readyToProcess`, it seems we won't seen this?